src/test/java/com/wikia/webdriver/testcases/discussions/ReportingPostTests.java

```diff
@@ -486,99 +469,106 @@ public void moderatorOnMobileCanDeleteReportedPostOnPostDetailsPage() {
+    userCannotReReportPostOnPostDetailPage();
+  }
+
+  private void userCannotReReportPostOnPostDetailPage() {
+    final PostEntity.Data data = createAndReportPostRemotelyAsFirstUser();
     validatePostRemotelyAsDiscussionsModerator(data);
 
-    PostDetailsPage page = new PostDetailsPage().open(data.getId());
-    PostEntity postEntity = page.getPost().findPostById(data.getId());
-    Assertion.assertFalse(postEntity.isReported(), NO_REPORTED_INDICATOR_ON_POST_MESSAGE);
-    Assertion.assertFalse(isReportPostOptionAvailableFor(postEntity), NO_REPORT_POST_OPTION_MESSAGE);
+    final PostDetailsPage page = openPostDetailsPageAndWaitUntilLoaded(data.getId());
+    final PostEntity postEntity = page.getPost().findPostById(data.getId());
+    assertFalse(postEntity.isReported(), NO_REPORTED_INDICATOR_ON_POST_MESSAGE);
+    assertFalse(isReportPostOptionAvailableFor(postEntity), NO_REPORT_POST_OPTION_MESSAGE);
   }
 
   @Test(groups = "discussions-loggedInUsersDesktopReporting")
   @Execute(asUser = User.USER)
   @InBrowser(browser = Browser.FIREFOX, browserSize = DESKTOP_RESOLUTION)
   public void userOnDesktopCannotReReportPostOnUserPostsPage() {
-    PostEntity.Data data = createAndReportPostRemotelyAsFirstUser();
+    userCannotReReportPostOnUserPostPage();
+  }
+
+  private void userCannotReReportPostOnUserPostPage() {
+    final PostEntity.Data data = createAndReportPostRemotelyAsFirstUser();
     validatePostRemotelyAsDiscussionsModerator(data);
 
-    PostEntity postEntity = new UserPostsPage().open(data.getAuthorId()).getPost().findPostById(data.getId());
-    Assertion.assertFalse(postEntity.isReported(), NO_REPORTED_INDICATOR_ON_POST_MESSAGE);
-    Assertion.assertFalse(isReportPostOptionAvailableFor(postEntity), NO_REPORT_POST_OPTION_MESSAGE);
+    final UserPostsPage userPostsPage = openUserPostsAndWaitUntilLoaded(data.getAuthorId());
+    final PostEntity postEntity = userPostsPage.getPost().findPostById(data.getId());
+    assertFalse(postEntity.isReported(), NO_REPORTED_INDICATOR_ON_POST_MESSAGE);
+    assertFalse(isReportPostOptionAvailableFor(postEntity), NO_REPORT_POST_OPTION_MESSAGE);
   }
 
   @Test(groups = "discussions-loggedInUsersDesktopReporting")
   @Execute(asUser = User.USER)
   @InBrowser(browser = Browser.FIREFOX, browserSize = DESKTOP_RESOLUTION)
   public void userOnDesktopCanNotSeeDeletedPostOnPostsListPage() {
-    PostEntity.Data data = createAndReportAndDeletePostRemotely();
+    final PostEntity.Data data = createAndReportAndDeletePostRemotely();
+    final PostsListPage page = openPostListPageAndWaitUntilLoaded();
+    final PostEntity postEntity = page.getPost().findPostById(data.getId());
+    assertNull(postEntity, NOT_VISIBLE_DELETED_POST_MESSAGE);
+  }
 
-    PostEntity post = openPostsListPage().getPost().findPostById(data.getId());
-    Assertion.assertNull(post, NOT_VISIBLE_DELETED_POST_MESSAGE);
+  private PostsListPage openPostListPageAndWaitUntilLoaded() {
```

Comment:
can you please move `openPostListPageAndWaitUntilLoaded` and `openPostDetailsPageAndWaitUntilLoaded` definitions to be next to other helper methods, below all test methods? It took me some time to find them in this 800+ lines file